src/plugins/intel_gpu/src/runtime/kernels_cache.hpp

```diff
@@ -106,6 +109,8 @@ class kernels_cache {
     }
     std::vector<kernel_id> add_kernels_source(std::vector<std::shared_ptr<kernel_string>> kernel_sources, bool dump_custom_program = false);
     void compile();
+    void save(BinaryOutputBuffer& ob) const;
+    void load(BinaryInputBuffer& ib);
```

Comment:
Why do we need to save/load kernels cache? Why you don't just save/load kernel in primitive_impl?

Reply:
Currently, up to 8 kernels are compiled at once. So up to 8 different kernels share the same kernel binary. For this reason, if it dumps kernel binaries in primitive_impl, it will store the same kernel binary up to 8 times. For this reason, the kernel binaries are saved to and loaded from kernels_cache instead of primitive_impl.